### PR TITLE
Enforce description size limits (per-skill 1024, per-plugin 15K)

### LINKED ISSUE #222: Measure/limit "description" material
"description" is read in to determine whether to load a skill (and all its tokens) or not.

It needs to clearly describe what the skill does and when to load and when not to load (so not so small it can't do that) but not be so large it wastes tokens. Also, users may apply a limit to the total size of "description" across all skills in a plugin and ignore anything beyond that.

We should have the infra
1) reject skills with "description" size beyond a limit. eg ., 1024 characters was a limit mentioned, we could choose smaller.
2) reject skills that take the combined size of descriptions in a single plugin beyond a limit. 15K characters was mentioned, we could choose smaller.

cc @timheuer seem right?

### PR DESCRIPTION
**Pull request by danmoseley**
Adds infrastructure to enforce description size limits per #222:

**Per-skill limit (1024 chars):**
- SkillProfiler flags DescriptionTooLong and warns
- ValidateCommand blocks with FailureKind = "description_too_long" (survives --verdict-warn-only)

**Per-plugin aggregate limit (15,000 chars):**
- New eng/validate-descriptions.ps1 checks both per-skill and aggregate limits
- New CI workflow validate-descriptions.yml runs on PRs touching plugins/**
- build.ps1 also enforces locally

**Current state:** all plugins pass (max skill: 769 chars, max plugin: 5,765 chars).

**Tests:** 3 new C# unit tests + 6 PowerShell integration tests covering boundary cases for both limits.

Closes #222

### CODE REVIEW

File: eng/validate-descriptions.ps1 [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+$ErrorActionPreference = 'Stop'
+
+if (-not $PluginsDir) {
+    $RepoRoot = Split-Path $PSScriptRoot -Parent
+    $PluginsDir = Join-Path $RepoRoot 'plugins'
+}
+
+function Get-SkillDescription([string]$FilePath) {
+    $content = Get-Content $FilePath -Raw
+    if ($content -notmatch '(?s)^---\r?\n(.*?)\r?\n---') {
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply eng/validate-descriptions.ps1]: Fixed — now using `(?ms)^---\r?\n(.*?)\r?\n^---\s*$` for robust delimiter matching.

File: eng/validate-descriptions.ps1 [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    # Block scalar (description: >- or description: > or description: | etc.)
+    if ($yaml -match '(?sm)^description:\s*[>|]-?\s*\r?\n((?:[ \t]+.*(?:\r?\n|$))*)') {
+        $lines = $Matches[1] -split '\r?\n' | ForEach-Object { $_.Trim() } | Where-Object { $_ }
+        return ($lines -join ' ')
+    }
+    # Double-quoted
+    if ($yaml -match 'description:\s*"((?:[^"\\]|\\.)*)"') {
+        return $Matches[1]
+    }
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply eng/validate-descriptions.ps1:43]: Addressed — script now uses `ConvertFrom-Yaml` when available (powershell-yaml module) for accurate YAML parsing, with the regex approach as fallback.

File: .github/workflows/validate-descriptions.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Validate Description Sizes
+
+on:
+  pull_request:
+    paths:
+      - 'plugins/**'
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply .github/workflows/validate-descriptions.yml:6]: Fixed — added `eng/validate-descriptions.ps1` and `.github/workflows/validate-descriptions.yml` to the `pull_request.paths` list.

File: agentic-workflows/dotnet-msbuild/build.ps1 [hunk 4]
```diff
@@ -39,10 +40,22 @@ foreach ($dir in $skillDirs) {
...
+
+    if ($description.Length -gt 1024) {
+        Write-Host "❌ $($dir.Name): Description too long ($($description.Length) chars, max 1024)" -ForegroundColor Red
+        $errors++
+    }
+}
+
+if ($totalDescChars -gt 15000) {
+    Write-Host "❌ Plugin aggregate description too long ($totalDescChars chars, max 15000)" -ForegroundColor Red
+    $errors++
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply agentic-workflows/dotnet-msbuild/build.ps1]: Addressed — removed duplicated limits from build.ps1 and delegated to `eng/validate-descriptions.ps1` as the single source of truth.

File: eng/tests/validate-descriptions.tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    param(
+        [hashtable[]]$Plugins
+    )
+    $root = Join-Path ([System.IO.Path]::GetTempPath()) "desc-test-$([guid]::NewGuid().ToString('N').Substring(0,8))"
+    foreach ($plugin in $Plugins) {
+        foreach ($skill in $plugin.Skills) {
+            $skillDir = Join-Path $root $plugin.Name 'skills' $skill.Name
+            New-Item -Path $skillDir -ItemType Directory -Force | Out-Null
+            $desc = $skill.Description
+            $content = "---`nname: $($skill.Name)`ndescription: `"$desc`"`n---`n# $($skill.Name)`nContent."
```
[the remark this answers is not in the document]
Comment (danmoseley) [reply eng/tests/validate-descriptions.tests.ps1]: Fixed — test fixture now escapes backslashes, double quotes, and newlines before embedding in YAML.

**Comment by danmoseley** [comment]
Yeah offline convo with @dbreshears with our proposal about maintainers. meanwhile - you're an invaluable person!

**Comment by danmoseley** [comment]
closing in favor of superset PR https://github.com/dotnet/skills/pull/238
